packages/shared-core/src/utils.ts

```diff
@@ -104,3 +119,96 @@ export function isSupportedUserSearch(query: SearchFilters) {
   }
   return true
 }
+
+/**
+ * Processes the filter config. Filters are migrated from
+ * SearchFilter[] to SearchFilterGroup
+ *
+ * If filters is not an array, the migration is skipped
+ *
+ * @param {LegacyFilter[] | SearchFilterGroup} filters
+ */
+export const processSearchFilters = (
+  filters: LegacyFilter[] | SearchFilterGroup | undefined
+): SearchFilterGroup | undefined => {
+  if (!filters) {
+    return
+  }
```

Comment:
Ideally would prefer to see this check pushed out to the caller and the type signature of this function to have the `undefined`s removed.

Reply:
As this is in the JS realm it needs to protect against the possibility of undefined, as we can't really protect against it in the JS code.